src/security/builtin_plugins/authentication/src/auth_utils.c

```diff
@@ -74,7 +74,7 @@ char *get_certificate_subject_name(X509 *cert, DDS_Security_SecurityException *e
 dds_time_t get_certificate_expiry(const X509 *cert)
 {
   assert(cert);
-  ASN1_TIME *asn1 = X509_get_notAfter(cert);
+  const ASN1_TIME *asn1 = X509_get0_notAfter(cert);
```

Comment:
Wouldn't it be better to change all uses of  `X509_get_notAfter` and `X509_get_notBefore` to the `get0` variant (or none of them)?